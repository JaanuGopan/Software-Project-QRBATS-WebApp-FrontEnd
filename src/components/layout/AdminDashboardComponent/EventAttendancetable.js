import React, { useState, useEffect } from "react";
import "../../../pages/AdminDashboard/AdminDashboard.css";
import html2canvas from "html2canvas";
import jsPDF from "jspdf";

const EventAttendancetable = ({
  search,
  attendanceList,
  handleGeneratePDF,
}) => {
  const [attendance, setAttendance] = useState([]);
  const [selectedAttendance, setSelectedAttendance] = useState(null);

  useEffect(() => {
    // Update events whenever eventList prop changes
    setAttendance(attendanceList);
  }, [attendanceList]);

  const handleAttendanceClick = (attendance) => {
    setSelectedAttendance(attendance);
  };

  return (
    <div className="tableDesign">
      <table className="student-tableArrangement">
        <thead>
          <tr>
            <th>No</th>
            <th className="expand">Student Name</th>
            <th>Registration Number</th>
            <th>Attendance Date</th>
            <th>Attendance Time</th>
          </tr>
        </thead>

        <tbody>
          {attendance
            .filter(
              (attendance) =>
                attendance.studentName
                  .toLowerCase()
                  .includes(search.toLowerCase()) ||
                attendance.indexNumber
                  .toLowerCase()
                  .includes(search.toLowerCase())
            )
            .map((attendance, index) => (
              <tr
                key={index}
                className={
                  selectedAttendance === attendance
                    ? "selected-row"
                    : "event-row"
                }
              >
                <td>{index + 1}</td>
                <td>{attendance.studentName}</td>
                <td>{attendance.indexNumber}</td>
                <td>{attendance.attendedDate}</td>
                <td>{attendance.attendedTime}</td>
              </tr>
            ))}
        </tbody>
      </table>
    </div>
  );
};

export default EventAttendancetable;
