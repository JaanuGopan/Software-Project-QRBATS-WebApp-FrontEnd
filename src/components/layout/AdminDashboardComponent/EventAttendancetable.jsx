import React, { useState, useEffect } from 'react';
import '../../../pages/AdminDashboard/AdminDashboard.css';
import html2canvas from 'html2canvas';
import jsPDF from 'jspdf';
import { FaCheck } from 'react-icons/fa';
import { AiOutlineClose } from 'react-icons/ai';

const EventAttendanceTable = ({ search, attendanceList, handleGeneratePDF }) => {
  const [attendance, setAttendance] = useState([]);
  const [selectedAttendance, setSelectedAttendance] = useState(null);

  useEffect(() => {
    // Update events whenever eventList prop changes
    setAttendance(attendanceList);
  }, [attendanceList]);

  const handleAttendanceClick = (attendance) => {
    setSelectedAttendance(attendance);
  };

  return (
    <div className="tableDesign">
      <table className="student-tableArrangement">
        <thead>
          <tr>
            <th>No</th>
            <th className="expand">Student Name</th>
            <th>Registration Number</th>
            <th>Attendance Date</th>
            <th>Attendance Time</th>
            <th>Attendance Status</th>
          </tr>
        </thead>

        {attendance.filter(
          (attendance) =>
            attendance.studentName.toLowerCase().includes(search.toLowerCase()) ||
            attendance.indexNumber.toLowerCase().includes(search.toLowerCase())
        ).length > 0 ? (
          <tbody>
            {attendance
              .filter(
                (attendance) =>
                  attendance.studentName.toLowerCase().includes(search.toLowerCase()) ||
                  attendance.indexNumber.toLowerCase().includes(search.toLowerCase())
              )
              .map((attendance, index) => (
                <tr
                  key={index}
                  className={selectedAttendance === attendance ? 'selected-row' : 'event-row'}
                >
                  <td>{index + 1}</td>
                  <td>{attendance.studentName}</td>
                  <td>{attendance.indexNumber}</td>
                  <td>{attendance.attendedDate}</td>
                  <td>{attendance.attendedTime}</td>
                  <td>
                    {attendance.attendanceStatus ? (
                      <FaCheck color="green" />
                    ) : (
                      <AiOutlineClose color="red" />
                    )}
                  </td>
                </tr>
              ))}
          </tbody>
        ) : (
          <tbody className="nodata">No Data Available</tbody>
        )}
      </table>
    </div>
  );
};

export default EventAttendanceTable;
