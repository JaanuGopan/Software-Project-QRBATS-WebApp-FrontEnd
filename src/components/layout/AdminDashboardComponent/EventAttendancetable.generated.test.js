import renderer from 'react-test-renderer';
import React, { useState, useEffect } from "react";

import html2canvas from "html2canvas";
import jsPDF from "jspdf";
import EventAttendancetable from "./EventAttendancetable";

jest.mock("../../../pages/AdminDashboard/AdminDashboard.css");
jest.mock("html2canvas");
jest.mock("jspdf");

const renderTree = tree => renderer.create(tree);
describe('<EventAttendancetable>', () => {
  it('should render component', () => {
    expect(renderTree(<EventAttendancetable 
    />).toJSON()).toMatchSnapshot();
  });
  
});