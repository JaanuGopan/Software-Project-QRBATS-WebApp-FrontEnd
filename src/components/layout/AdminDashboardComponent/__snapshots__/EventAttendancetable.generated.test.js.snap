// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventAttendancetable> should render component 1`] = `
<div
  className="tableDesign"
>
  <table
    className="student-tableArrangement"
  >
    <thead>
      <tr>
        <th>
          No
        </th>
        <th
          className="expand"
        >
          Student Name
        </th>
        <th>
          Registration Number
        </th>
        <th>
          Attendance Date
        </th>
        <th>
          Attendance Time
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
